### ISSUE TITLE
WorkerConfig for runtime: dotnet-isolated not found

### ISSUE THREAD

**Issue by brennfoster** [Azure/azure-functions-dotnet-worker#821]
I don't see a specific issue for my use case so I'm creating a new one. Azure function app does not start after deployment. This works fine locally. Only when I deploy to Azure do I see this problem.

According to documentation, my configuration settings are correct:
![image](https://user-images.githubusercontent.com/419283/156857658-be3dd1d8-6339-4a41-96ca-65f4c7397e66.png)

- Function runtime version: ~4
- Function worker runtime: dotnet-isolated
- .NET version: 6.0
- OS: linux
- Deployment method: run from package
- Deployment tool: terraform
- Deployment source: Azure pipelines

Error message:
> System.InvalidOperationException : WorkerConfig for runtime: dotnet-isolated not found
>     at async Microsoft.Azure.WebJobs.Script.Workers.Rpc.RpcFunctionInvocationDispatcher.InitializeAsync(IEnumerable`1 functions,CancellationToken cancellationToken) at /src/azure-functions-host/src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs : 236
>     at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
>     at async Microsoft.Azure.WebJobs.Script.ScriptHost.InitializeAsync(CancellationToken cancellationToken) at /src/azure-functions-host/src/WebJobs.Script/Host/ScriptHost.cs : 328
>     at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
>     at async Microsoft.Azure.WebJobs.Script.ScriptHost.StartAsyncCore(CancellationToken cancellationToken) at /src/azure-functions-host/src/WebJobs.Script/Host/ScriptHost.cs : 265
>     at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
>     at async Microsoft.Extensions.Hosting.Internal.Host.StartAsync(CancellationToken cancellationToken)
>     at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
>     at async Microsoft.Azure.WebJobs.Script.WebHost.WebJobsScriptHostService.UnsynchronizedStartHostAsync(ScriptHostStartupOperation activeOperation,Int32 attemptCount,JobHostStartupMode startupMode) at /src/azure-functions-host/src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs : 309

Fails at: https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs

```
var workerConfig = _workerConfigs.Where(c => c.Description.Language.Equals(_workerRuntime, StringComparison.InvariantCultureIgnoreCase)).FirstOrDefault();
if (workerConfig == null && (functions == null || functions.Count() == 0))
{
    // Only throw if workerConfig is null AND some functions have been found.
    // With .NET out-of-proc, worker config comes from functions.
    throw new InvalidOperationException($"WorkerConfig for runtime: {_workerRuntime} not found");
}
```

I've seen multiple posts about this on Stack Overflow and GitHub but so far I cannot find any resolution or work around.

**Comment by kshyju** [comment]
@brennfoster Could you try deploying using [function core tools](https://docs.microsoft.com/en-us/azure/azure-functions/functions-run-local?tabs=v4%2Cwindows%2Ccsharp%2Cportal%2Cbash#publish) and see that works?

**Comment by brennfoster** [comment]
I tried to publish using function core tools but I get this error:

`Your app has an unknown FUNCTIONS_WORKER_RUNTIME defined 'dotnet-isolated'. Only dotnet, node, python, powershell, custom are supported.
You can pass --force to update your Azure app with 'None' as a 'FUNCTIONS_WORKER_RUNTIME'`

**Comment by brennfoster** [comment]
I tried supplying '--force' when publishing using function core tools but the runtime still won't load any functions

**Comment by kshyju** [comment]
@brennfoster Yea,  you need to use the `--force` flag.(ex: `func azure functionapp publish your_app_name --force`)

Does it make any difference if you try again to publish (using core tools)?

**Comment by kshyju** [comment]
Just to confirm, are you seeing the value of `FUNCTIONS_WORKER_RUNTIME` setting to be "None" when you check "Settings->Configuration->Application settings" in azure portal for your app?

**Comment by brennfoster** [comment]
Correct

**Comment by kshyju** [comment]
We can investigate why that happened when published using CLI. Could you try manually changing the value to "dotnet-isolated" via portal and see that fixes the issue?

**Comment by brennfoster** [comment]
Still seeing the error. When I initially deploy this error says "Microsoft.Azure.WebJobs.Script: WorkerConfig for runtime: None not found.". After manually changing the setting I see the below error:
![image](https://user-images.githubusercontent.com/419283/159034261-385acce1-a1b6-4eb3-ba12-a65ed2b5033c.png)

**Comment by brennfoster** [comment]
Oops, wrong screen shot:
![image](https://user-images.githubusercontent.com/419283/159034593-c2d481d3-febf-48f3-bd52-fcbc7a2113e4.png)

**Comment by brennfoster** [comment]
Any updates here?

**Comment by brennfoster** [comment]
I wanted to get ahead of the curve being that .NET 7 looks like it will only run out of process, but we are getting to a point where we can't wait on this anymore. Should we not be using isolated mode for linux environments?

Please advise.

**Comment by kshyju** [comment]
@brennfoster We are currently investigating this issue.  Our current suspicion is that this is caused by a corrupted deployment package.

When you tried to publish, did the publish operation succeed? For example if you used VS publish option, did you see a "successfully published" message?

**Comment by olehgerus** [comment]
I tried direct publish to the Azure (VS 2022, Linux app) and it is failed (`System.AggregateException: One or more errors occurred. ---> Microsoft.WebTools.Shared.Exceptions.WebToolsException: Publish has encountered an error. .`)

I also tried 'Docker Container Registry' publish and it was succeed. But the function doesn't start with the mentioned `WorkerConfig for runtime: dotnet-isolated not found` error

**Comment by brennfoster** [comment]
Publishing from Visual Studio 2022 gives me this error:
`System.AggregateException: One or more errors occurred. ---> Microsoft.WebTools.Shared.Exceptions.WebToolsException: Publish has encountered an error. We were unable to determine the cause of the error. Check the output log for more details.`

**Comment by kshyju** [comment]
Thanks @olehgerus @brennfoster for confirming. I will keep this thread updated as we make progress on the investigation.

cc @balag0 FYI

**Comment by brennfoster** [comment]
Interestingly, I got this to work by upgrading to v3 of terraforms azurerm provider and using the new modules for function apps.

**Comment by kshyju** [comment]
FYI @eliaslopezgt

**Comment by fabiocav** [comment]
Leaving this open as we'll need to continue investigating the deployment issue with the folks @kshyju has tagged above.

**Comment by nathanarca** [comment]
I dont know what happening, but in last friday i had the same problem, bu now he dont back happening. had any update on this weekend?

**Comment by dasokume** [comment]
I had the same issue, but after updating my Nuget packages the issue is gone.
I hope that can be helpful for someone
![image](https://user-images.githubusercontent.com/35662765/164002731-930e85a2-43bd-4d12-b59b-cb09cba2919b.png)
https://docs.microsoft.com/en-us/azure/azure-functions/functions-versions?tabs=isolated-process%2Cv4&pivots=programming-language-csharp

**Comment by sulmanm** [comment]
I am having same issue.
Created .NET 6.0 Isolated Function (with Http trigger) in Visual Studio 2022. Then When I run it in VS 2022, same error:

![image](https://user-images.githubusercontent.com/24681343/164212847-3c78faf7-c5dc-4c3b-8952-27cbe0be7a50.png)

All the required nuget packages are installed 
![image](https://user-images.githubusercontent.com/24681343/164212994-a9c34566-eefb-40a4-ae99-d25acb37390d.png)

is there any else is also required to run the '.NET 6.0 Isolated functions' locally in Visual Studio 2022?

**Comment by kshyju** [comment]
@sulmanm Could you share your repro(which fails locally as well) as a GH repo?

**Comment by sulmanm** [comment]
GitHub Repo can be found here: [FunctionAppTest](https://github.com/sulmanm/FunctionAppTest)

**Comment by kshyju** [comment]
@sulmanm Thanks! I was able to run this project without any issue.

**Comment by sulmanm** [comment]
For me it does not work and fails with error: WorkerConfig for runtime: dotnet-isolated not 

I am running Visual Studio 17.1.5 on Window 10

![image](https://user-images.githubusercontent.com/24681343/164648170-1b1be6a3-4652-47b7-81a0-9eb04a585404.png)

**Comment by kshyju** [comment]
@sulmanm Could you try in a second machine? If still reproducing, please open a new issue and continue the conversation there. Let's keep this issue for azure deployment behavior. Thanks!

**Comment by shahabganji** [comment]
For future references:  I had the same issue when I am trying to dockerize the application:

By adding the `ENV FUNCTIONS_WORKER_RUNTIME=dotnet-isolated` to my docker file. This usually should reside on your `local.settings.json` or on the configuration of the azure function app. 

my docker file: 

```
FROM mcr.microsoft.com/dotnet/sdk:6.0 AS installer-env

# Build requires 3.1 SDK
COPY --from=mcr.microsoft.com/dotnet/core/sdk:3.1 /usr/share/dotnet /usr/share/dotnet

COPY . /src/dotnet-function-app
RUN cd /src/dotnet-function-app && \
    mkdir -p /home/site/wwwroot && \
    dotnet publish *.csproj --output /home/site/wwwroot

# To enable ssh & remote debugging on app service change the base image to the one below
# FROM mcr.microsoft.com/azure-functions/dotnet:4-appservice
FROM mcr.microsoft.com/azure-functions/dotnet-isolated:4
ENV AzureWebJobsScriptRoot=/home/site/wwwroot \
    AzureFunctionsJobHost__Logging__Console__IsEnabled=true

ENV FUNCTIONS_WORKER_RUNTIME=dotnet-isolated

COPY --from=installer-env ["/home/site/wwwroot", "/home/site/wwwroot"]
```

**Comment by JosefBackovsky** [comment]
We have same exception „WorkerConfig for runtime: dotnet-isolated not found„ in logs from time to time. Azure function runs on Windows Operating System with „dotnet-isolated“ properly set in „FUNCTIONS_WORKER_RUNTIME“. Deployed via Azure DevOPS pipeline.

Exception appliers in small batch of 6 - 8 for two minutes. Then dissipater and system works properly.

**Comment by wesselkranenborg** [comment]
> We have same exception „WorkerConfig for runtime: dotnet-isolated not found„ in logs from time to time. Azure function runs on Windows Operating System with „dotnet-isolated“ properly set in „FUNCTIONS_WORKER_RUNTIME“. Deployed via Azure DevOPS pipeline.
> 
> Exception appliers in small batch of 6 - 8 for two minutes. Then dissipater and system works properly.

Same issue here indeed. We deploy to a staging slot, do a warm-up of our eventhub triggered function and when we swap the slot to production. During swap we see the following errors in application insights and they recover themselves but it causes downtime in the processing of events coming from event hubs during that period.
<img width="739" alt="image" src="https://user-images.githubusercontent.com/4964604/166650529-85181f0f-0898-4bda-80e0-93c208ae90a9.png">

We have set the appsetting `FUNCTIONS_WORKER_RUNTIME=dotnet-isolated` to both the production and staging slot.

**Comment by tsukashusan** [comment]
I too had the same problem.
However, I solved this problem by setting the Platform setting to 64-bit.
When Functions are created in Azure portal, they seem to be set to 32-bit by default.

<img width="634" alt="image" src="https://user-images.githubusercontent.com/25189375/169214165-0fe38853-243d-42db-a71d-7c31291e1faf.png">

**Comment by wesselkranenborg** [comment]
We use this setting in our ARM template, so with us it didn't solve the problem.

<img width="278" alt="image" src="https://user-images.githubusercontent.com/4964604/169475538-741048dc-662f-4685-bff0-a1b43d718c92.png">

**Comment by mahesh-maximus** [comment]
I changed the `"use32BitWorkerProcess": false` , but still seeing the error "WorkerConfig for runtime: dotnet-isolated not found" after each deployment.

This error throws from below code in the [azure-functions-host](https://github.com/Azure/azure-functions-host/blob/2360468596d0c6fb0e79b2315ec6b889dcefe38e/src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs#L255)
![image](https://user-images.githubusercontent.com/6337584/171626270-bb09e9eb-9366-40f7-a0e3-ce9bf990f7cd.png)

**Comment by mahesh-maximus** [comment]
@kshyju i do the deployment using the azure pipeline and it is successful, i think this issue is happening just after the deployment

**Comment by lavibhatnagar** [comment]
Does this issue is fixed? I am facing same issue when trying to deploy from azure devops pipeline. Deployment shows it is succeeded. but can see this error 'Microsoft.Azure.WebJobs.Script: WorkerConfig for runtime: dotnet-isolated not found.' in the portal. As well as navigating to functions i can see the warning 'azure functions runtime is unreachable'. 

I am using windows function app.
![image](https://user-images.githubusercontent.com/5319479/177624070-240629f3-ac6c-4d9f-a147-b3645645f62a.png)

As mentioned above i have tried changing 32 bit platform to 64-bit still same error. Though i am able successfully publish from visual studio 2022.

Can anyone please help me in this?

**Comment by Joffreyvl** [comment]
Has anybody found a solution to this problem?

**Comment by lavibhatnagar** [comment]
Hi,

I am able to fixed this issue by adding and deploying test function in my project. I created empty project and deployed it without any functions.

Adding and deploying test function in my project solution fixed issue for me.

**Comment by DavidZidar** [comment]
@kshyju What is the proper way to create and configure a Linux app service so this works? I have been trying and trying and keep getting this error.

This is what I have been trying to do in many different variations using the CLI and the portal:
1. Create function app on a brand new App Service plan `az functionapp create --functions-runtime 4 --os-type Linux --runtime dotnet --runtime-version 6 [...]`
2. Set linuxFxVersion `az functionapp config set --linux-fx-version "DOTNET-ISOLATED|6.0" [...]`
3. Set FUNCTIONS_WORKER_RUNTIME to `dotnet-isolated`
4. And now it's broken

It doesn't matter if I try to deploy any code or not, I only get this error no matter what I do. The Kudu site is up and running and it accepts my zip package and correctly mounts it if I try to deploy, the docker logs seem to indicate that it also starts the function app, but the host is non functional.

**Comment by DavidZidar** [comment]
@kshyju I have managed to get a successful deployment once or twice using `func azure functionapp publish` but never using zip deploy.

The one thing I notice that is different is that when I use `func azure functionapp publish` the zip file that's placed in `data/SitePackages` contains a folder named `.azurefunctions` with some assemblies in it. When I use zip deploy the zip I am uploading does contain the same folder but for some reason it is removed from the zip that is then stored in `data/SitePackages`.

Everything else is exactly identical as far as I can tell.

**Comment by DavidZidar** [comment]
Oof, I finally got zip deploy to work, it turns out it was my zip that was the problem after all. Because the `.azurefunctions` folder has a dot in the name it is regarded as a hidden folder on Linux, which caused the folder to be omitted on my Linux build agent but was included like it was supposed to on Windows when I was validating my code and comparing my zip archive so I thought it was included when in fact it was not.

It would be nice with some validation at deploy time, if necessary files are missing from the zip then it probably shouldn't be accepted. Or at the very least it should be validated and logged in the Docker logs.

**Comment by Joffreyvl** [comment]
@DavidZidar so how did you get it to work? Did you change de folder name?

**Comment by DavidZidar** [comment]
@Joffreyvl No, in my case the problem was that I was building in Linux and did `zip -r foo.zip *` which I thought included all files and folders but it unfortunately omits hidden files and folders. My solution was to call `zip -r foo.zip .` instead to bypass shell parameter expansion and tell zip to compress everything in the current folder.

I did try to deploy from Windows as well which produced a correct zip as far as I can tell but that didn't work, maybe my App Service instance was already corrupted at that time.

**Comment by RbsTechOnline** [comment]
I am having a similar problem from devops, trying to publish to consumption windows, isolated, .net 6. works from visual studio. Zip deployment are the same. configuration looks the same, DOES NOT work from `AzureFunctionApp@1` in a release pipeline

```
steps:
- task: AzureFunctionApp@1
  displayName: 'Azure Function App Deploy: test'
  inputs:
    azureSubscription: 'sp-devops'
    appType: functionApp
    appName: 'test'
    package: '$(System.DefaultWorkingDirectory)/Artifact/test.zip'
    appSettings: '-FUNCTIONS_WORKER_RUNTIME dotnet-isolated'
    deploymentMethod: runFromPackage

```
If have spent hours trying to tweak this in every way possible, comparing the difference between devops deploy and visual studio deploy, for the life of me this just doesn't work.....

Can i ask. Is .net 6 isolated officially supported yet ? or is it just the devops tooling stuck in a time warp ?

Do i need to escalate this to our support contract ?

**Comment by hbrotan** [comment]
I have the same problem. 

- Framework: `.NET 6`
- Function version: `v4`
- FUNCTIONS_WORKER_RUNTIME: `dotnet-isolated`

Deploying using bicep: 
```
    - name: 'Deploy Azure function'
      uses: azure/functions-action@v1
      with:        
        app-name: '${{ inputs.azure_func_name }}'
        package: ./${{inputs.azure_func_project_path}}
```
On startup, the host logs `0 functions loaded`, even if there are one HTTP trigger and one ServiceBusTrigger: 
![image](https://user-images.githubusercontent.com/5398896/182789175-6e5247c6-9c6a-4d77-a4ad-e5b4bb217994.png)

Exception:
![image](https://user-images.githubusercontent.com/5398896/182789529-b8e7dcf4-c72b-4dbd-a642-b03f43a8c444.png)

**Comment by zubaircse-bd** [comment]
I have the same problem.

Framework: .NET 6
Function version: v4
FUNCTIONS_WORKER_RUNTIME: dotnet-isolated
![image](https://user-images.githubusercontent.com/20569360/183023016-e8f2040d-018d-4ac8-9855-b4ebd75b08ff.png)

**Comment by hbrotan** [comment]
At least there seems to be a fix coming for this error when deploying without a payload, ref: fix by @kshyju here: https://github.com/Azure/azure-functions-host/pull/8589

But, I get this error even if I'm deploying to an existing Azure function. 
I've tried setting the linuxFxVersion = `dotnet-isolated|6.0`.

**Comment by Joffreyvl** [comment]
Hey everyone, we got it to work by using this configuration.

- Function runtime version: ~4
- Function worker runtime: dotnet-isolated
- .NET version: 6.0
- OS: linux
- Deployment method: run from package
- Deployment tool: bicep
- Deployment source: Azure pipelines

![image](https://user-images.githubusercontent.com/17174541/183031720-a57e6995-c0ab-4d37-b5be-5bbfa708a9d5.png)

![image](https://user-images.githubusercontent.com/17174541/183031629-06264337-d3b4-4bc8-aad1-c2b8893704ec.png)

We found the solution in the [official Microsoft documentation](https://docs.microsoft.com/en-us/azure/azure-functions/dotnet-isolated-process-guide).

![image](https://user-images.githubusercontent.com/17174541/183031943-88e9e83c-98b9-4594-9eae-cd401d2aecd8.png)

**Comment by hbrotan** [comment]
@Joffreyvl .NET version: 6.0 in combination with `DOTNET-ISOLATED|7.0` gives me errors of type: 
```
You must install or update .NET to run this application.

The following frameworks were found:
Framework: 'Microsoft.NETCore.App', version '6.0.0' (x64)

To install missing framework, download:
7.0.0-preview.6.22324.4 at [/usr/share/dotnet/shared/Microsoft.NETCore.App]
```
Are you sure it's working with .NET 6.0?

**Comment by Joffreyvl** [comment]
Documentation says, DOTNET-ISOLATED|7.0 but DOTNET-ISOLATED|6.0 works for me

**Comment by Dragsaw** [comment]
The following configuration works for us with bicep deployment: `linuxFxVersion` set to `DOTNET-ISOLATED|6.0`,
`FUNCTIONS_EXTENSION_VERSION` to  `~4` and `FUNCTIONS_WORKER_RUNTIME` to `dotnet-isolated`.

Unless a .net function code is deployed to the azure function app, the portal shows _"WorkerConfig for runtime: dotnet-isolated not found"_ error. The error goes away after a function using isolated process model is deployed, because if you look at the .net build output directory it includes a `worker.config.json` file and all necessary libraries.

The worker config comes from the .net function package. While there are no functions in the resource the portal will keep showing the _"dotnet-isolated not found"_ error.

**Comment by hbrotan** [comment]
@Dragsaw I think there are two issues here. One is when you deploy without any functions, which is fixed in https://github.com/Azure/azure-functions-host/pull/8589. 

But, the error also occurs when you deploy _with_ functions, and I still get this error even if setting `linuxFxVersion` to `DOTNET-ISOLATED|6.0`

**Comment by Dragsaw** [comment]
@hbrotan , I see. Unfortunately, I have not been able to make `dotnet-isolated|7.0` work, only `6.0`. 
When you're using `6.0`, are you able to check that `worker.config.json` file is present in the function package that is being deployed and that the contents are similar to this?
``` json
{
  "description": {
    "language": "dotnet-isolated",
    "extensions": [ ".dll" ],
    "defaultExecutablePath": "dotnet",
    "defaultWorkerPath": "MyFunctionApp.dll"
  }
}
```

Also, there shoulld be a file like `MyFunctionApp.runtimeconfig.json` with the similar settings:
``` json
{
  "runtimeOptions": {
    "tfm": "net6.0",
    "framework": {
      "name": "Microsoft.NETCore.App",
      "version": "6.0.0"
    }
  }
}
```
These two file are automatically generated during project build/publish, so if they were wrong then some project setting would have to be changed.

Besides that, if you are deploying to an existing Function App, I would suggest clearing the files in the storage account related to the function app. The service bugged out on me once and would not change any configurations until those files in the storage account were removed. However, I cannot say how safe this operation is or if it will have any side effects, most likely it is not recommended.

**Comment by hbrotan** [comment]
@Dragsaw Thanks for answering. I have the same `worker.config.json` and almost the same `MyFunctionApp.runtimeconfig.json` with the addition of: 
```
"configProperties": {
      "System.Reflection.Metadata.MetadataUpdater.IsSupported": false
    }
```
I think the `dotnet-isolated|7.0` is a dead end. It requires upgrading to .NET7 preview, and that should not be the solution to this issue.

**Comment by MarkiEC13** [comment]
I had the same issue and downgraded the following packages from the latest version and all started to work without any issues.
`<PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.2.0" />`
`<PackageReference Include="Microsoft.Azure.Functions.Worker" Version="1.5.0" />`

**Comment by hbrotan** [comment]
I tried adding a dummy HTTP trigger (anonymous auth), and the error occurred less, but the error still occurred.
This is the log for an app with 7 functions (ordered by timestamp desc): 

![image](https://user-images.githubusercontent.com/5398896/184139945-d3e306b2-cdf6-4611-a7bd-0d408f34ea40.png)

Because 1 function is found (the HTTP trigger), it will not fail.
If no functions are found, there are no config, and it will fail.

Seems like this is a startup issue, where the host is not able to detect the functions. (How can 7 functions be loaded, if only 1 function was found?).

Anyways, I hope the [fix ](https://github.com/Azure/azure-functions-host/pull/8589) from @kshyju might solve this issue.

**Comment by kshyju** [comment]
> But, the error also occurs when you deploy _with_ functions, and I still get this error even if setting `linuxFxVersion` to `DOTNET-ISOLATED|6.0`

@hbrotan Could you share your function app name & time of deployment so I can take a look at this behavior?

**Comment by hbrotan** [comment]
@kshyju Yes, that would be great, thanks! 
App name: `hmc-func-service-email-dev-we`
Time of deployment (UTC): `8/12/2022, 8:23:54.024 AM`

Screenshot trace:
![image](https://user-images.githubusercontent.com/5398896/184316249-c90319a7-0faa-48e4-a91c-9f7c3445d511.png)

Screenshot exception:
![image](https://user-images.githubusercontent.com/5398896/184316404-91779da0-68a6-4c67-97a5-c83e76abea47.png)

**Comment by hbrotan** [comment]
@kshyju Will you get a chance to look at this? Much appreciated.

**Comment by kshyju** [comment]
@hbrotan I looked into our logs and can see that, for role instance "6BADE189-637958780685568709", it appears that the deployment was not really successful (function app payload was not deployed correctly), so it had '0 functions found" messages logged. Can you double check the function app files are properly deployed?

Good news is, we have a fix coming out in the next few weeks to address this issue.

**Comment by hbrotan** [comment]
@kshyju Thanks for checking! It seems like this error comes and goes. And it seems like the functions are working in spite of the error, maybe because they are able to start after the initial error.

I believe there is a bug in the framework when loading functions metadata. 
How can you explain this behavior? (https://github.com/Azure/azure-functions-dotnet-worker/issues/821#issuecomment-1211968960) -> 1 functions found, but 7 functions loaded?
I believe the bug is that the framework is unable to detect all functions, even if they are there. Might be some race conditions? 

Anyway, I saw that you made a fix when deploying without functions. It might solve this (hopefully), but not the real issue which I believe is that functions are not found. 

If I run my project locally using Azure Function Core Tools (4.0.4679) `func start --verbose`:
```
[2022-08-18T10:55:46.065Z] Loading functions metadata
[2022-08-18T10:55:46.069Z] Reading functions metadata
[2022-08-18T10:55:46.070Z] Reading functions metadata
[2022-08-18T10:55:46.073Z] 0 functions found
[2022-08-18T10:55:46.075Z] 0 functions found
[2022-08-18T10:55:46.091Z] Reading functions metadata
[2022-08-18T10:55:46.143Z] 1 functions found
[2022-08-18T10:55:46.146Z] 2 functions loaded
```

It doesn't fail, but what's happening here with `0 functions found`? Why isn't the functions found? Is it the same issue when starting up after deploy, but then it will actually fail because no functions = no config?

**Comment by hbrotan** [comment]
@clpo Yes, the error will appear if 0 functions are found, because the config is part of the function. So various errors with the function might result in this.

@kshyju I know you're busy, but I'd really like a response to the behavior I described in my previous post (7 days ago). Thanks.

**Comment by kshyju** [comment]
@hbrotan Could you open a new issue with relevant details / repro steps, so that we can focus on that problem in that issue.

**Comment by lAnubisl** [comment]
I have Azure function where I was able to fix this issue somehow. Don't know how. Here is the [Terraform sample](https://gist.github.com/lAnubisl/24d551d348736d1b1852f4c486b65dc8) which looks correct but when it is deployed the function gets broken immediately (even without any real code in it). I have compared ARM templates of fixed and broken azure functions and have not found anything important different. Has anybody found a workaround how to make dotnet-isolated functions work?

**Comment by lAnubisl** [comment]
Here is [ARM Template](https://gist.github.com/lAnubisl/8d5ded50534b5f249c0b78eb9d55d34b) which is for Azure function creation. Immediately after deployment the function is broken.

**Comment by Dragsaw** [comment]
> Here is [ARM Template](https://gist.github.com/lAnubisl/8d5ded50534b5f249c0b78eb9d55d34b) which is for Azure function creation. Immediately after deployment the function is broken.

Are you referring to "WorkerConfig for runtime: dotnet-isolated not found" error? If you see it immediately after creating the Azure Function App then try deploying actual .Net function code there. That can hopefully resolve the problem. That is the case for me as I mentioned [before](https://github.com/Azure/azure-functions-dotnet-worker/issues/821#issuecomment-1206325815).

**Comment by lAnubisl** [comment]
> Are you referring to "WorkerConfig for runtime: dotnet-isolated not found" error? If you see it immediately after creating the Azure Function App then try deploying actual .Net function code there. That can hopefully resolve the problem. That is the case for me as I mentioned [before](https://github.com/Azure/azure-functions-dotnet-worker/issues/821#issuecomment-1206325815).

Probably the issue is somehow related to "Consumption Plan" case. Because once function resource is broken it is not possible to deploy a function there.

```
System.AggregateException: One or more errors occurred. ---> Microsoft.WebTools.Shared.Exceptions.WebToolsException: Publish has encountered an error. We were unable to determine the cause of the error. Check the output log for more details. 
   --- End of inner exception stack trace ---
---> (Inner Exception #0) Microsoft.WebTools.Shared.Exceptions.WebToolsException: Publish has encountered an error. We were unable to determine the cause of the error. Check the output log for more details. <---
```

And the uotput says The attempt to publish the ZIP file through https://func-................scm.azurewebsites.net/api/zipdeploy failed with HTTP status code Failed.

**Comment by kshyju** [comment]
The "**_WorkerConfig for runtime: dotnet-isolated not found_**" error appears when a dotnet-isolated function app is created, but no function payload was successfully deployed.  We recently made some changes to the host runtime, so that this scenario will not be considered an error case anymore. From host version [4.10.1](https://github.com/Azure/azure-functions-host/releases/tag/v4.10.1) onwards, you should be able to create a dotnet-isolated app without the payload deployed and no error will appear.

There has been improvements in the core tools and VS publish pipeline since this issue was originally created. So at this point, I believe this issue is fixed.

@brennfoster Could you try to repro the issue with latest version of tooling/host runtimes?
